pkg/e2e/helpers.go

```
@@ -53,8 +53,13 @@ func WaitForIngress(s *Sandbox, ing *v1beta1.Ingress) (*v1beta1.Ingress, error)
 		result := validator.Check(context.Background())
 		if result.Err == nil {
 			return true, nil
+		} else {
+			if expectUnreachable {
+				return false, nil
+			} else {
+				return true, fmt.Errorf("Unexpected error from validation: %v", result.Err)
```

return false, fmt.Errorf("Unexpected error from validation: %v", result.Err)? `Poll()` will return once there is an error.

---

I want it to return in this case. If expectUnreachable is false, meaning we expect no 404's/502's, then I want to return the error if there was one.

---

I see, that makes sense. I was mistakenly thinking expectUnreachable means we expect it to be always unreachable.